pkg/collector/corechecks/servicediscovery/systemprobe/module_linux.go

```diff
@@ -0,0 +1,135 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux
+
+package systemprobe
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"path"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/prometheus/procfs"
+
+	"github.com/DataDog/datadog-agent/cmd/system-probe/api/module"
+	"github.com/DataDog/datadog-agent/cmd/system-probe/config"
+	sysconfigtypes "github.com/DataDog/datadog-agent/cmd/system-probe/config/types"
+	"github.com/DataDog/datadog-agent/cmd/system-probe/utils"
+	"github.com/DataDog/datadog-agent/comp/core/telemetry"
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/servicediscovery/model"
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/servicediscovery/portlist"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+)
+
+const (
+	pathOpenPorts = "/open_ports"
+	pathGetProc   = "/procs/{pid}"
+)
+
+// ServiceDiscoveryModule is the service_discovery module factory.
+var ServiceDiscoveryModule = module.Factory{
```

Comment:
Apologies for the confusion. I think this type `ServiceDiscoveryModule` should stay in `cmd/system-probe/modules`. The reason is just to reduce confusion as to the possible modules in system-probe. Having them all defined in the same package is helpful.

The type `serviceDiscovery` type can stay in this package though. I may also suggest creating a constructor init function for `serviceDiscovery`(for example `func NewServiceDiscoveryModule() (module.Module, error)`) so that imports like `portlist` (since that is internal to `serviceDiscovery`) can remain in this package.

Happy to chat if you disagree/have questions.

Reply:
no problem! 
what you propose makes sense, created `NewServiceDiscoveryModule` in this package.